File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]
        common_prefix = strs[0]
        for word in strs[1:]:
            new_common = ''
            for i in range(min(len(common_prefix), len(word))):
                if common_prefix[i] == word[i]:
                    new_common += word[i]
                else:
                    break
            common_prefix = new_common
        return common_prefix

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = strs[0];
        for (int i = 1; i < strs.size(); i++) {
            string temp = "";
            for (int j = 0; j < strs[i].length(); j++) {
                if (ans[j] != strs[i][j]) break;
                temp.push_back(strs[i][j]);
            }
            ans = temp; 
        }
        return ans;
    }
};